docs/howto/features/private-nbgitpuller.md

```diff
@@ -0,0 +1,105 @@
+# Allow nbgitpuller to pull from private GitHub repos
+
+[nbgitpuller](https://github.com/jupyterhub/nbgitpuller) is very popularly
+used to pull repos from public GitHub repositories. We can also allow users
+to pull from *private* GitHub repositories, with [git-credential-helpers](https://github.com/yuvipanda/git-credential-helpers).
+
+## Configure the image
+
+The image used in the JupyterHub must have the [git-credential-helpers](https://pypi.org/project/git-credential-helpers/)
+package installed from PyPI. Unfortunately this is currently not installed in
+any common upstream images, so this will only work with custom images.
+
+## Setup GitHub app
+
+`git-credentials-helper` uses a [GitHub App](https://docs.github.com/en/developers/apps)
+to pull private repos. So you first need to create a GitHub app for each hub that wants
+to pull private repos as static content.
+
+1. Create a [GitHub app in the 2i2c org](https://github.com/organizations/2i2c-org/settings/apps/new).
+
+2. Give it a descriptive name (such as '<hub-name> private repo access') and description, as users will see this when authorizing
+   access to their private repos.
+
+3. Disable webhooks (uncheck the 'Active' checkbox under 'Webhooks'). All other
+   textboxes can be left empty.
+
+4. Under 'Repository permissions', select 'Read' for 'Contents'.
+
+5. Under 'Where can this GitHub App be installed?', select 'Any account'. This will
+   enable users to push to their own user repositories or other organization repositories,
+   rather than just the 2i2c repos.
+
+6. Create the application with the 'Create GitHub app' button.
+
+7. Copy the numeric 'App id' from the app info page you should be redirected to.
+
+8. Create a new private key for authentication use with the `Generate a private key`
+   button. This should download a private key file, that you should delete after
+   putting it in the appropriate config (in the next step)
+
+## Helm values configuration
+
+Now, we configure our user servers to authenticate themselves with GitHub using
+the app we just created.
+
+1. Tell `git` to use our GitHub app for authenticating when cloning from GitHub.
+
+   ```yaml
+   jupyterhub:
+     singleuser:
+       extraFiles:
+         gitconfig:
+           mountPath: /etc/gitconfig
+           stringData: |
+             [credential "https://github.com"]
+             helper = !git-credential-github-app --app-key-file /etc/github/github-app-private-key.pem --app-id <app-id>
+          useHttpPath = true
```

Comment:
I'm slightly confused about the `useHttpPath` parameter. I'm guessing it is either part of `stringData` and needs further indentation like so:

```suggestion
            useHttpPath = true
```

Or it is a new key and should be like so:

```suggestion
          useHttpPath: true
```

Since you typed `=`, I'm going to assume the former first.

Reply:
Yep, it had wrong indent! Fixed, thanks for catching this.